test/requests/baremetal/node_tests.rb

```diff
@@ -0,0 +1,93 @@
+require "test_helper"
+
+describe "Fog::Baremetal[:openstack] | Baremetal node requests" do
+  before do
+    @baremetal = Fog::Baremetal[:openstack]
+    @node_format = {
+      'instance_uuid'   => Fog::Nullable::String,
+      'maintenance'     => Fog::Boolean,
+      'power_state'     => Fog::Nullable::String,
+      'provision_state' => Fog::Nullable::String,
+      'uuid'            => String,
+      'links'           => Array
+    }
+
+    @detailed_node_format = {
+      'instance_uuid'          => Fog::Nullable::String,
+      'maintenance'            => Fog::Boolean,
+      'power_state'            => Fog::Nullable::String,
+      'provision_state'        => Fog::Nullable::String,
+      'uuid'                   => String,
+      'created_at'             => String,
+      'updated_at'             => Fog::Nullable::String,
+      'chassis_uuid'           => Fog::Nullable::String,
+      'console_enabled'        => Fog::Boolean,
+      'driver'                 => String,
+      'driver_info'            => Hash,
+      'extra'                  => Hash,
+      'instance_info'          => Hash,
+      'last_error'             => Fog::Nullable::String,
+      'maintenance_reason'     => Fog::Nullable::String,
+      'properties'             => Hash,
+      'provision_updated_at'   => Fog::Nullable::String,
+      'reservation'            => Fog::Nullable::String,
+      'target_power_state'     => Fog::Nullable::String,
+      'target_provision_state' => Fog::Nullable::String,
+      'links'                  => Array
+    }
+  end
+
+  describe "success" do
+    it "#list_nodes" do
+      @baremetal.list_nodes.body.must_match_schema('nodes' => [@node_format])
+    end
+
+    it "#list_nodes_detailed" do
+      @baremetal.list_nodes_detailed.body.
+        must_match_schema('nodes' => [@detailed_node_format])
+    end
+
+    before do
```

Comment:
Similar note as before - I would like to see `before` defined in the beginning of `describe` block.